components/organisms/Chat/ChatMessages/SoqlResult.tsx

```diff
@@ -0,0 +1,72 @@
+import { isObject } from "radash";
+import { SalesforceQueryResult } from "@/shared/types/salesforceTypes";
+
+type ResultTableProps = {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  error: any;
+  result: SalesforceQueryResult;
+  instanceUrl?: string;
+};
+
+export const SoqlResult = ({
+  error,
+  result,
+  instanceUrl,
+}: ResultTableProps) => {
+  if (error) {
+    return (
+      <h1 className={"font-semibold text-red-500 font-mono"}>
+        Error running query, Retrying request.
+      </h1>
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const records: any[] = result?.records;
```

Comment:
This smells - typing should be done when passing in

Reply:
Its weird tho cos records can have any schema (totally dependant on the salesforce object being referred to). So rn, i just render every key when displaying the table and check if its an object when passing it in